tests/simulationbase.py

```diff
@@ -173,55 +182,112 @@ def _simulate(self, task_params: dict, tmpdir, expected_frames: list):
             time.sleep(3)
 
             task_id = 'test_task_id123'
-            req_task_work_dir = req_work_dir / task_id
-            req_task_work_dir.mkdir()
-            req_task_resources_dir = req_task_work_dir / constants.RESOURCES_DIR
-            req_task_resources_dir.mkdir()
-            req_task_net_resources_dir = \
-                req_task_work_dir / constants.NETWORK_RESOURCES_DIR
-            req_task_net_resources_dir.mkdir()
-            req_task_results_dir = req_task_work_dir / constants.RESULTS_DIR
-            req_task_results_dir.mkdir()
-            req_task_net_results_dir = \
-                req_task_work_dir / constants.NETWORK_RESULTS_DIR
-            req_task_net_results_dir.mkdir()
-
-            prov_task_work_dir = prov_work_dir / task_id
-            prov_task_work_dir.mkdir()
-            prov_task_net_resources_dir = \
-                prov_task_work_dir / constants.NETWORK_RESOURCES_DIR
-            prov_task_net_resources_dir.mkdir()
+            req_task_work_dir, req_task_resources_dir, \
+                req_task_net_resources_dir, req_task_results_dir, \
+                req_task_net_results_dir = \
+                self._make_req_dirs(req_work_dir, task_id)
+
+            prov_task_work_dir, prov_task_net_resources_dir = \
+                self._make_prov_dirs(prov_work_dir, task_id)
 
             self._put_cube_to_resources(req_task_resources_dir)
 
             self._create_task(requestor, task_id, task_params)
 
             for _ in range(task_params['subtasks_count']):
 
-                subtask_id, subtask_params = \
-                    self._get_next_subtask(requestor, task_id)
-                assert subtask_params['resources'] == [0]
-
-                self._copy_resources_from_requestor(
+                self._do_subtask(
```

Comment:
it does not return true, it asserts at the end of `_do_subtask()`

Reply:
I meant that it should instead return the verdict to be asserted later, since we may wanna test invalid subtasks and assert failure in some other test

Reply:
ah, clear, fixed